### PR TITLE
Updates 04.12

### PR DESCRIPTION
**Pull request by Apollon77**
* (@Apollon77) Shows a progress indicator when deleting controller nodes
* (@Apollon77) Cuts names and labels to 32 or 64 characters as needed by Matter
* (@Apollon77) Improves error handling on devices and bridges
* (@Apollon77) Clear storage when removing a bridged device
* (@Apollon77) Processes changed objects with a 5s delay to prevent too many changes at once
* (@Apollon77) Fixes version determination

### CODE REVIEW

File: src/lib/DeviceManagement.ts [hunk 1]
```diff
@@ -266,7 +266,35 @@ class MatterAdapterDeviceManagement extends DeviceManagement<MatterAdapter> {
             }
         }
 
-        await this.adapter.controllerNode?.decommissionNode(node.nodeId);
+        const progress = await context.openProgress(this.#adapter.t('Unpairing node...'), { label: '0%' });
+
+        // Start an interval that normally covers 30s and with each update the number gets slower increased for the percentage
+        let finished = false;
+        let timeout: NodeJS.Timeout | undefined = undefined;
```
Comment (foxriver76) [inline src/lib/DeviceManagement.ts:273]: Does not need to be initialized

File: src/lib/DeviceManagement.ts [hunk 2]
```diff
@@ -266,7 +266,35 @@ class MatterAdapterDeviceManagement extends DeviceManagement<MatterAdapter> {
...
+                return;
+            }
+            timeout = setTimeout(updateProgress, 1000);
+        };
+        timeout = setTimeout(updateProgress, 1000);
+
+        try {
+            await this.adapter.controllerNode?.decommissionNode(node.nodeId);
+        } catch (error) {
+            await context.showMessage(this.#adapter.t('Error during unpairing: ') + error);
```
Comment (foxriver76) [inline src/lib/DeviceManagement.ts:290]: Error as object?

File: src/main.ts [hunk 3]
```diff
@@ -89,6 +90,13 @@ export class MatterAdapter extends utils.Adapter {
     #nodeReSyncInProgress = new Set<string>();
     #closing = false;
     #version: string = '0.0.0';
+    #objectProcessQueue = new Array<{
```
Comment (foxriver76) [inline src/main.ts:93]: Any reason for this uncommon syntax?

**Review by foxriver76 — APPROVED** [review]
For what is this queue needed tbh?

**Comment by Apollon77** [comment]
Re Queue: The minimum needed is a delay in processing the changed objects if someone in the UI klicks very fast because else re-initializations could overlap and lead to strange behavior. SO I thought it is a good idea to also queue that all if someone changes multiple devices, so that logs are easier to read and we do not overload "smaller systems"

File: src/main.ts (hunk 3 above)
Comment (Apollon77) [reply src/main.ts:93]: for me new Array is common in the meantime and easy because in sync with Set/Map and such comparable types

**Pull request merged**